confusing error message when configuration is corrupted

Edit by Tau: 
This is not something that can easily be recovered from, but there should be an error message that explains the situation.



On Vanilla OS 2.0, running abroot gives me the following error

```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0xa8 pc=0x120ac15]

goroutine 1 [running]:
main.main()
	/__w/ABRoot/ABRoot/main.go:63 +0x795
```

Although I don't know why this happened, I assume that my abroot config is corrupted. I don't think I can recover from this issue, since I would need `abroot` to change the config, and I have no other means of accessing the host-shell to change the config manually, do I?

Creating a config in any of the location in the repo's Readme does not help, since I'm not creating it in the host shell!?

I cannot even tell which version of abroot I'm using and rollbacks are also not possible.

I think this [line](https://github.com/Vanilla-OS/ABRoot/blob/ba0f914488191147b052b68f5018ebfb64ff297a/main.go#L63) needs to be changed to not depend on a config file, otherwise it might make Vanilla OS unusable. At least in my case, I opted for reinstalling from scratch.

--- a/settings/config.go
+++ b/settings/config.go
@@ -74,25 +74,41 @@ var (
 
 func init() {
 
+	foundConfig := false
+
 	// system path
 	viper.SetConfigFile(CnfPathSystem)
 	err := viper.ReadInConfig()
 	if err != nil {
 		fmt.Fprintln(os.Stderr, "could not read config file in /usr", err)
+	} else {
+		foundConfig = true
 	}
 
 	// admin path
 	viper.SetConfigFile(CnfPathAdmin)
 	err = viper.MergeInConfig()
 	if err != nil {
-		fmt.Fprintln(os.Stderr, "could not read config file in /usr", err)
+		fmt.Fprintln(os.Stderr, "could not read config file in /etc", err)
+	} else {
+		foundConfig = true
 	}
 
 	// dev paths
 	viper.SetConfigFile(CnfPathDev1)
-	viper.MergeInConfig()
+	err = viper.MergeInConfig()
+	if err == nil {
+		foundConfig = true
+	}
 	viper.SetConfigFile(CnfPathDev2)
-	viper.MergeInConfig()
+	err = viper.MergeInConfig()
+	if err == nil {
+		foundConfig = true
+	}
+
+	if !foundConfig {
+		panic("could not find an intact config")
+	}
 
 	// VanillaOS specific defaults for backwards compatibility
 	viper.SetDefault("updateInitramfsCmd", "lpkg --unlock && /usr/sbin/update-initramfs -u && lpkg --lock")

